pkg/install/v1alpha1/cstor_volume.go

```diff
@@ -621,7 +621,12 @@ spec:
     Calculate the replica count
     Add as many poolUid to resources as there is replica count
     */}}
-    {{- $poolUids := keys .ListItems.cvolPoolList.pools | randomize }}
+    {{- $replicaAffinityName:= .TaskResult.creategetpvc.replicaAntiAffinity -}}
+    {{- if ne $replicaAffinityName  "none" -}}
```

Comment:
Make use of ternary logic.
Make use of empty function from https://github.com/openebs/maya/blob/master/pkg/template/template.go#L40
Expose this as a go template function.

```go
// IfNotNil returns the second argument if first 
// argument is not nil else it returns the first 
// argument i.e. nil
func IfNotNil(this, then interface{}) interface{} {
  if !empty(this) {
    return then
  }
  return this
}
```
```yaml
{{- $lblSelector := printf "openebs.io/replica-anti-affinity=%s" $val | IfNotNil $val -}}
```

Reply:
added